src/main/java/net/kaczmarzyk/spring/data/jpa/web/WebRequestProcessingContext.java

```diff
@@ -18,21 +18,38 @@
 import java.lang.annotation.Annotation;
 import java.util.Map;
 
+import net.kaczmarzyk.spring.data.jpa.utils.PathVariableResolver;
 import org.springframework.core.MethodParameter;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.context.request.NativeWebRequest;
 
-import net.kaczmarzyk.spring.data.jpa.utils.PathVariableResolver;
 import net.kaczmarzyk.spring.data.jpa.utils.QueryContext;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.HandlerMapping;
 
 import javax.servlet.http.HttpServletRequest;
 
 /**
+ * <p>
  * Provides information about Controller/method and WebRequest being processed.
  * It is a wrapper around low-level Spring classes, which provides easier access to e.g. path variables.
- * 
+ * When the request attributes don't contain information about extracted path variables,
+ * the path variables are extracted using a fallback approach (which has some limitations, eg. it does not support global prefixes).
+ * </p>
+ * <br>
+ * Old approach takes only controllers mapping and tries to match it against request URI.
+ * As it requires paths to be exact the same it fails and throws exception
+ * <p>
+ * example controllers mapping: <br>
+ * /user/{userId} <br>
+ *
+ * example request URI with global api prefix that would fail for old approach: <br>
```

Comment:
I still don't undestand what "global api prefix" is. Add link/reference or example to how it would be configured